SimpleNotePad.java

```diff
@@ -0,0 +1,222 @@
+import java.awt.print.PrinterJob;
+import java.io.*;
+public class SimpleNotePad extends JFrame implements ActionListener{
+    private static final int MAX_NUMBER_OF_RECENT_FILES = 5;
+
+    private JMenuBar menuBar = new JMenuBar();
+    private JMenu fileMenu = new JMenu("File");
+    private JMenu editMenu = new JMenu("Edit");
+    private JTextPane userDataTextPane = new JTextPane();
+    private JMenuItem openMenuItem = new JMenuItem("Open");
+    private JMenuItem recentMenuItem = new JMenu("Recent");
+    private JMenuItem newFileMenuItem = new JMenuItem("New File");
+    private JMenuItem saveFileMenuItem = new JMenuItem("Save File");
+    private JMenuItem printFileMenuItem = new JMenuItem("Print File");
+    private JMenuItem copyMenuItem = new JMenuItem("Copy");
+    private JMenuItem pasteMenuItem = new JMenuItem("Paste");
+    private JMenuItem replaceMenuItem = new JMenuItem("Replace");
+    private File[] recentlyOpenedFiles = new File[MAX_NUMBER_OF_RECENT_FILES];
+
+    public SimpleNotePad() {
+        setTitle("A Simple Notepad Tool");
+        buildFileMenu();
+        buildEditMenu();
+        menuBar.add(fileMenu);
+        menuBar.add(editMenu);
+        setJMenuBar(menuBar);
+        add(new JScrollPane(userDataTextPane));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    }
+
+    //create a File menu with open, print, save, new file and recent items in it
+    private void buildFileMenu(){
+        fileMenu.add(openMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(printFileMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(saveFileMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(newFileMenuItem);
+        fileMenu.addSeparator();
+        fileMenu.add(recentMenuItem);
+
+        newFileMenuItem.addActionListener(this);
+        newFileMenuItem.setActionCommand("new");
+        saveFileMenuItem.addActionListener(this);
+        saveFileMenuItem.setActionCommand("save");
+        printFileMenuItem.addActionListener(this);
+        printFileMenuItem.setActionCommand("print");
+        recentMenuItem.addActionListener(this);
+        recentMenuItem.setActionCommand("recent");
+        openMenuItem.addActionListener(this);
+        openMenuItem.setActionCommand("open");
+    }
+
+    // create Edit menu with copy, paste and replace items in it
+    private void buildEditMenu(){
+        editMenu.add(copyMenuItem);
+        editMenu.add(pasteMenuItem);
+        editMenu.add(replaceMenuItem);
+
+        copyMenuItem.addActionListener(this);
+        copyMenuItem.setActionCommand("copy");
+        pasteMenuItem.addActionListener(this);
+        pasteMenuItem.setActionCommand("paste");
+        replaceMenuItem.addActionListener(this);
+        replaceMenuItem.setActionCommand("replace");
+    }
+
+    // build a list of recent open items
+    private void buildRecentMenu(){
```

Comment:
If you want, you could move methods from here to below into another class file and it could shorten this file and make it easier to read